nix/autofirma/create-autofirma-cert

```diff
@@ -0,0 +1,103 @@
+_autofirma_dir="$1"
+_autofirma_ca="${_autofirma_dir}/AutoFirma_ROOT.cer"
+_autofirma_pfx="${_autofirma_dir}/autofirma.pfx"
+_cert_days="3650"
+_cert_cn="AutoFirma ROOT"
+
+function _make_ca_config {
+  cat << EOF > "${_temp_dir}/openssl.cnf"
+[ ca ]
+default_ca=CA_autofirma
+[ CA_autofirma ]
+dir=${_temp_dir}
+new_certs_dir=\$dir
+database=\$dir/index.txt
+serial=\$dir/serial
+crlnumber=\$dir/crlnumber
+default_days=${_cert_days}
+default_crl_days=30
+default_md=sha256
+preserve=no
+x509_extensions=usr_cert
+email_in_dn=no
+copy_extensions=copy
+[ policy_ca ]
+countryName=optional
+stateOrProvinceName=optional
+localityName=optional
+organizationName=optional
+organizationalUnitName=optional
+commonName=supplied
+emailAddress=optional
+[ req ]
+default_bits=4096
+x509_extensions=v3_ca
+distinguished_name=req_distinguished_name
+[ req_distinguished_name ]
+commonName_default=${_cert_cn}
+[ usr_cert ]
+basicConstraints=CA:FALSE
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid:always,issuer:always
+subjectAltName=IP:127.0.0.1
+[ v3_ca ]
+basicConstraints=critical,CA:TRUE
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid:always,issuer:always
+keyUsage=cRLSign,digitalSignature,keyCertSign,keyEncipherment,dataEncipherment
+extendedKeyUsage=serverAuth,clientAuth,anyExtendedKeyUsage
+EOF
+touch "${_temp_dir}/index.txt"
+echo "01" > "${_temp_dir}/crlnumber"
+}
```

Comment:
Idea: consider using of bash’s here document `<<-EOF` [variant](https://www.gnu.org/software/bash/manual/bash.html#Here-Documents), which allows indenting the payload, like this:
```
function foo {
cat <<-EOF > some_file
<tab><tab>payload line number 1
<tab><tab>payload line number 2
<tab>EOF
<tab>some_other_command
}
```
The structure would be more easily parsed by the human brain.
**CAVEAT EMPTOR**: tabs must be used, spaces don’t work with this feature.

Reply:
Great! I'll also indent the rest of the script so it is easier to read.